packages/components/src/unit-control/utils.js

```diff
@@ -11,135 +11,117 @@ import { Platform } from '@wordpress/element';
+		a11yLabel: _x( 'ems', 'Relative to parent font size (em)' ),
 	},
-	{
+	rem: {
 		value: 'rem',
 		label: isWeb ? 'rem' : __( 'Relative to root font size (rem)' ),
 		default: '',
-		a11yLabel: __( 'Relative to root font size (rem)' ),
+		a11yLabel: _x( 'rems', 'Relative to root font size (rem)' ),
 	},
-	{
+	vw: {
 		value: 'vw',
 		label: isWeb ? 'vw' : __( 'Viewport width (vw)' ),
 		default: '',
 		a11yLabel: __( 'Viewport width (vw)' ),
 	},
-	{
+	vh: {
 		value: 'vh',
 		label: isWeb ? 'vh' : __( 'Viewport height (vh)' ),
 		default: '',
 		a11yLabel: __( 'Viewport height (vh)' ),
 	},
-	{
+	vmin: {
 		value: 'vmin',
 		label: isWeb ? 'vmin' : __( 'Viewport smallest dimension (vmin)' ),
 		default: '',
 		a11yLabel: __( 'Viewport smallest dimension (vmin)' ),
 	},
-	{
+	vmax: {
 		value: 'vmax',
 		label: isWeb ? 'vmax' : __( 'Viewport largest dimension (vmax)' ),
 		default: '',
 		a11yLabel: __( 'Viewport largest dimension (vmax)' ),
 	},
-	{
+	ch: {
 		value: 'ch',
 		label: isWeb ? 'ch' : __( 'Width of the zero (0) character (ch)' ),
 		default: '',
 		a11yLabel: __( 'Width of the zero (0) character (ch)' ),
 	},
-	{
+	ex: {
 		value: 'ex',
 		label: isWeb ? 'ex' : __( 'x-height of the font (ex)' ),
 		default: '',
 		a11yLabel: __( 'x-height of the font (ex)' ),
 	},
-	{
+	cm: {
 		value: 'cm',
 		label: isWeb ? 'cm' : __( 'Centimeters (cm)' ),
 		default: '',
 		a11yLabel: __( 'Centimeters (cm)' ),
 	},
-	{
+	mm: {
 		value: 'mm',
 		label: isWeb ? 'mm' : __( 'Millimeters (mm)' ),
 		default: '',
 		a11yLabel: __( 'Millimeters (mm)' ),
 	},
-	{
+	in: {
 		value: 'in',
 		label: isWeb ? 'in' : __( 'Inches (in)' ),
 		default: '',
 		a11yLabel: __( 'Inches (in)' ),
 	},
-	{
+	pc: {
 		value: 'pc',
 		label: isWeb ? 'pc' : __( 'Picas (pc)' ),
 		default: '',
 		a11yLabel: __( 'Picas (pc)' ),
 	},
-	{
+	pt: {
 		value: 'pt',
 		label: isWeb ? 'pt' : __( 'Points (pt)' ),
 		default: '',
 		a11yLabel: __( 'Points (pt)' ),
 	},
-];
+};
+
+/**
+ * An array of all available CSS length units.
+ */
+export const ALL_CSS_UNITS = Object.values( allUnits );
 
 /**
  * Units of measurements. `a11yLabel` is used by screenreaders.
  */
 export const CSS_UNITS = [
-	{ value: 'px', label: 'px', default: 0, a11yLabel: __( 'pixels' ) },
-	{ value: '%', label: '%', default: 10, a11yLabel: __( 'percent' ) },
-	{
-		value: 'em',
-		label: 'em',
-		default: 0,
-		a11yLabel: _x( 'ems', 'Relative to parent font size (em)' ),
-	},
-	{
-		value: 'rem',
-		label: 'rem',
-		default: 0,
-		a11yLabel: _x( 'rems', 'Relative to root font size (rem)' ),
-	},
-	{
-		value: 'vw',
-		label: 'vw',
-		default: 10,
-		a11yLabel: __( 'viewport widths' ),
-	},
-	{
-		value: 'vh',
-		label: 'vh',
-		default: 10,
-		a11yLabel: __( 'viewport heights' ),
-	},
+	{ ...allUnits.px, default: 0 },
+	{ ...allUnits.percent, default: 0 },
+	{ ...allUnits.em, default: 0 },
+	{ ...allUnits.rem, default: 0 },
+	{ ...allUnits.vw, default: 10 },
+	{ ...allUnits.vh, default: 10 },
```

Comment:
Why the defaults are different here compared to the array above?

Reply:
They were always like that so I just ported them without digging in.

Backtracing it now, it looks like these defaults were added in https://github.com/WordPress/gutenberg/pull/20888/files#diff-4161524a5c509693e7a2412feb7767fa2c69d841e870fc80a35d33ac3fbeb28c but they were never actually used anywhere, the cover block was defining its own defaults in https://github.com/WordPress/gutenberg/pull/20888/files#diff-98e8fbed8e10d45fee8dfcb0373d733aa1335f7cbf0a5dd883739b86fae0fd7e

Since then things have changed and in the https://github.com/WordPress/gutenberg/pull/31822 refactor we added `defaultValues` in all blocks that needed them.

I suppose we can remove them from here, I don't think they are actually used anywhere :thinking:

Reply:
Let's clean this out 👍

Reply:
Done :+1: